diff -urN php52-5.2.17/ext/openssl/config0.m4 php52-5.2.17-openssl/ext/openssl/config0.m4
--- php52-5.2.17/ext/openssl/config0.m4	2007-01-07 19:38:22.000000000 +0100
+++ php52-5.2.17-openssl/ext/openssl/config0.m4	2013-01-08 15:16:27.617292699 +0100
@@ -17,6 +17,7 @@
   fi
 
   AC_CHECK_LIB(ssl, DSA_get_default_method, AC_DEFINE(HAVE_DSA_DEFAULT_METHOD, 1, [OpenSSL 0.9.7 or later]))
+  AC_CHECK_LIB(crypto, X509_free, AC_DEFINE(HAVE_DSA_DEFAULT_METHOD, 1, [OpenSSL 0.9.7 or later]))
 
   PHP_SETUP_OPENSSL(OPENSSL_SHARED_LIBADD, 
   [
