Index: php52-5.2.17~precise~ppa19/ext/gd/config.m4
===================================================================
--- php52-5.2.17~precise~ppa19.orig/ext/gd/config.m4    2015-09-20 08:40:12.110492599 +0000
+++ php52-5.2.17~precise~ppa19/ext/gd/config.m4 2015-09-20 09:11:49.622545723 +0000
@@ -210,7 +210,7 @@
   if test "$PHP_FREETYPE_DIR" != "no"; then
 
     for i in $PHP_FREETYPE_DIR /usr/local /usr; do
-      if test -f "$i/include/freetype2/freetype/freetype.h"; then
+      if test -f "$i/include/freetype2/freetype/freetype.h" || test -f "$i/include/freetype2/freetype.h";  then
         FREETYPE2_DIR=$i
         FREETYPE2_INC_DIR=$i/include/freetype2
         break
